app/routes/analyzer.tsx

```diff
@@ -735,6 +746,83 @@ function AbilityPointsDetails({
   );
 }
 
+/**
+ * From an array of Main abilities, create a map of <Ability, number>
+ *  that describes the number of Ability chunks required to replace the main ability on a piece of gear.
+ *
+ * Extra processing is required for Main abilities that are primary slot-only abilities,
+ *  as they are comprised of 3 stackable ability chunks at a lower ability chunk count than usual.
+ */
+function getAbilityChunksMap(mainAbilities: AbilityWithUnknown[]) {
+  const abilityChunksMap: MainAbilityChunks = new Map();
+  for (let i = 0; i < mainAbilities.length; i++) {
+    const mainAbility = mainAbilities[i];
+    if (typeof mainAbility !== "undefined" && mainAbility !== "UNKNOWN") {
+      const primarySlotOnlyAbilityRef = abilities.filter(
+        (a) => a.name === mainAbility && a.abilityChunkTypesRequired.length > 0
+      );
+
+      // Primary-slot-only item that can have ability chunks put on it
+      if (primarySlotOnlyAbilityRef.length === 1) {
+        const primaryAbility = primarySlotOnlyAbilityRef[0];
+        if (typeof primaryAbility !== "undefined") {
```

Comment:
I end up having to use a lot of these `if (typeof variableName !== "undefined")` checks. Is there a better way of doing this in my current use cases? TypeScript forces me to add these checks for type safety, and I'm just wondering if there's a less cluttered way of doing these checks.

Reply:
You are inside a loop. So I think it would make it quite a bit cleaner to just do: 

```ts
if (!variableName) continue

// logic continues here
```